modules/sitemaps/sitemaps.php

```diff
@@ -1,792 +1,1865 @@
-			echo "<$key>";
-			jetpack_print_xml_tag( $value );
-			echo "</$key>";
-		} else {
-			echo "<$key>" . esc_html( $value ) . "</$key>";
-		}
+	/**
+	 * Construct a new Jetpack_Sitemap_Manager.
+	 *
+	 * @since 4.5.0
+	 */
+	public function __construct() {
+		// Register post types for data storage.
+		add_action(
+			'init',
+			array( $this, 'callback_action_register_post_types' )
+		);
+
+		// Add callback for sitemap URL handler.
+		add_action(
+			'init',
+			array( $this, 'callback_action_catch_sitemap_urls' )
+		);
+
+		// Add generator to wp_cron task list.
+		$this->schedule_sitemap_generation();
+
+		// Add sitemap to robots.txt.
+		add_action(
+			'do_robotstxt',
+			array( $this, 'callback_action_do_robotstxt' ),
+			20
+		);
+
+		/*
+		 * Module parameters are stored as options in the database.
+		 * This allows us to avoid having to process all of init
+		 * before serving the sitemap data. The following actions
+		 * process and store these filters.
+		 */
+
+		// Process filters and store post_types for inclusion in sitemap.
+		add_action(
+			'init',
+			array( $this, 'callback_action_filter_sitemap_post_types' ),
+			999
+		);
+
+		// Process filters and store location string for sitemap.
+		add_action(
+			'init',
+			array( $this, 'callback_action_filter_sitemap_location' ),
+			999
+		);
+
+		// Process filters and store skipped post IDs.
+		add_action(
+			'init',
+			array( $this, 'callback_action_filter_sitemap_skip_post' ),
+			999
+		);
+
+		return;
 	}
-}
 
-/**
- * Convert an array to a SimpleXML child of the passed tree.
- *
- * @module sitemaps
- *
- * @param array $data array containing element value pairs, including other arrays, for XML contruction.
- * @param SimpleXMLElement $tree A SimpleXMLElement class object used to attach new children.
- *
- * @return SimpleXMLElement full tree with new children mapped from array.
- */
-function jetpack_sitemap_array_to_simplexml( $data, &$tree ) {
-	$doc_namespaces = $tree->getDocNamespaces();
-
-	foreach ( $data as $key => $value ) {
-		// Allow namespaced keys by use of colon in $key, namespaces must be part of the document
-		$namespace = null;
-		if ( false !== strpos( $key, ':' ) && 'image' != $key ) {
-			list( $namespace_prefix, $key ) = explode( ':', $key );
-			if ( isset( $doc_namespaces[ $namespace_prefix ] ) ) {
-				$namespace = $doc_namespaces[ $namespace_prefix ];
-			}
+	/**
+	 * Callback to register sitemap post types for data storage.
+	 *
+	 * @access public
+	 * @since 4.5.0
+	 */
+	public function callback_action_register_post_types() {
+		/**
+		 * Helper method for registering sitemap data post types uniformly.
+		 *
+		 * @since 4.5.0
+		 *
+		 * @param string $type_name The name of the new post type. Must be at most 20 characters long.
+		 * @param string $label The label of the new post type.
+		 * @param string $slug The slug of the new post type.
+		 */
+		function register_sitemap_data( $type_name, $label, $slug ) {
```

Comment:
Scratch that- sitemaps are now stored in a custom database table, so this code is not relevant anymore.